app/src/main/java/com/globant/harrypotterapp/activity/HouseActivity.kt

```diff
@@ -0,0 +1,43 @@
+package com.globant.harrypotterapp.activity
+
+import android.app.Activity
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.viewpager2.widget.ViewPager2
+import com.globant.harrypotterapp.R
+import com.globant.harrypotterapp.adapter.ViewPagerAdapter
+import com.globant.harrypotterapp.databinding.ActivityHouseBinding
+import com.google.android.material.tabs.TabLayout
+import com.google.android.material.tabs.TabLayoutMediator
+
+class HouseActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityHouseBinding
+    private var houseName: String? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityHouseBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        houseName = intent.getStringExtra(HOUSE_NAME)
+        binding.activityHouseAppBarTitle.text = houseName
```

Comment:
you can make a private method called initUI or initialize or something like that and put there these two lines:
```
binding.activityHouseAppBarTitle.text = houseName
initViewPager()
```

Reply:
Refactored!